torch/_inductor/kernel/flex_attention.py

```diff
@@ -1113,78 +1275,149 @@ def bwd_dkdv_inner(
     # BLOCK_N1 must be a multiple of BLOCK_M1, otherwise the code wouldn't work.
     tl.static_assert(BLOCK_N1 % BLOCK_M1 == 0)
     hi = sparse_q_num_blocks * SPARSE_Q_MULTIPLE
-    for start_m in range(0, hi):
-        # Load LSE before computing qk to reduce pipeline stall.
-
-        qT = tl.load(qT_ptrs)
-        lse = tl.load(LSE + offs_m1)
-        qkT = tl.dot(k, qT)
-        if not PRESCALE_QK:
-            qkT *= SM_SCALE
-        # ~~~~~~~~~~~~~~~~~~~ Apply score modification  ~~~~~~~~~~~~~~~~~~~
+
+    if not IS_DIVISIBLE:
+        if hi >= 1:
```

Comment:
Might be cleaner to move this condition to only apply to the cleanup block?

Reply:
Do you mean sth like:
```
if IS_DIVISIBLE or (not IS_DIVISIBLE and hi = 0):
    run regular for loop
else:
    run for loop until hi -1 and run the last compute block
```

I think it would be cleaner, but I'm a bit worry on the perf may hurt divisible case. Since ```IS_DIVISIBLE``` is kernel options passed before generating the kernel, Triton should compile the version beforehand. But ```hi``` is computed from accessing the ```kv_num_blocks```, which is decided when the tensor is ready. I will run a benchmark to check if this really hurt.

Reply:
Verified, it doesn't hurt perf, will refactor it.

Reply:
I tried the other way like:
```
for start_n in range(block_n_start, block_n_end - 1):
    fwd_compute_block_mn(...)
if not IS_DIVISIBLE:
    fwd_compute_block_mn(..., is_last_block=True)
else:
    fwd_compute_block_mn(...)
```

But I found the divisible case's perf degraded, so I decided to keep my original way. It probably because my current way leverages better pipeline process?